### ISSUE TITLE
Unable to download HA generation script

### ISSUE THREAD

**Issue by marksullivancrowd** [rancher/rancher#4372]
Rancher Version:
v1.0.1-rc2 CATTLE_RANCHER_SERVER_IMAGE=v1.0.1-rc2
Docker Version:
Server Version: 1.10.1
OS: 
boot2docker virtualbox, MacOSX mavericks 
Steps to Reproduce: 
```
server:
  image: rancher/server:v1.0.1-rc2
  ports:
   - 8080:8080
  environment:
   - CATTLE_DB_CATTLE_MYSQL_HOST=mysql
   - CATTLE_DB_CATTLE_MYSQL_NAME=cattle
   - CATTLE_DB_CATTLE_MYSQL_PORT=3306
   - CATTLE_DB_CATTLE_USERNAME=cattle
   - CATTLE_DB_CATTLE_PASSWORD=cattle
  links:
   - mysql:mysql
mysql:
  image: percona:5.6
  environment:
   - MYSQL_ALLOW_EMPTY_PASSWORD='yes'
   - MYSQL_DATABASE=cattle
   - MYSQL_USER=cattle
   - MYSQL_PASSWORD=cattle
  log_driver: json-file
  log_opt:
    max-size: 10m
  volumes:
   - /projects/dbs/rancher/mysql:/var/lib/mysql
```

proceed to admin/ha fill out hostname 3 hosts, and provide key cert and bundle for ssl cert, click generate script, click download script....
Results: 
server_1 | 2016/04/11 23:53:49 http: proxy error: net/http: request canceled
Expected:
file download to browser

**Comment by vincent99** [comment]
There's a weird bug where the download fails the very first time you use it.. as a workaround if you try again it should work

**Comment by marksullivancrowd** [comment]
gets to a stage where you cant get back to generating the HA config

you get this 
![screen shot 2016-04-12 at 01 43 59](https://cloud.githubusercontent.com/assets/546141/14446632/1395966a-0050-11e6-9a30-66b1f248f42e.png)
then this
![screen shot 2016-04-12 at 01 46 02](https://cloud.githubusercontent.com/assets/546141/14446717/c0921564-0050-11e6-8954-2fdbf26e3100.png)

and any time you go back to the HA page it assumes you have downloaded the script, an you need to restart the rancher/server container

here's some audit logs, ive removed the certs and the host name

```
request api.haConfig.createscript
{
  "certChain": "-----BEGIN CERTIFICATE----------END CERTIFICATE-----",
  "zookeeperClientPort": 2181,
  "swarmPort": 2367,
  "zookeeperLeaderPort": 3888,
  "ppHttpPort": 81,
  "httpPort": 80,
  "httpsPort": 443,
  "redisPort": 6375,
  "clusterSize": 3,
  "hostRegistrationUrl": "https://{valid url}”,
  "cert": "-----BEGIN CERTIFICATE-----",
  "ppHttpsPort": 444,
  "zookeeperQuorumPort": 2888,
  "key": "-----BEGIN RSA PRIVATE KEY-----",
  "httpEnabled": true,
  "swarmEnabled": true
}
response
{
  "id": "c32aa999-9d47-4071-990f-14a287783d18",
  "type": "error",
  "links": {},
  "actions": {},
  "status": 500,
  "code": "ScriptFailed",
  "message": "Failed to create script",
  "detail": null
}

```

is there a main log file in the container or a debug env i could enable?
so i could catch this
http: proxy error: net/http: request canceled

**Comment by marksullivancrowd** [comment]
from /var/lib/cattle/logs/cattle-debug.log
when i click download ha script

```
2016-04-12 01:26:29,148 DEBUG [:] [] [] [] [qtp639313753-91] [i.c.p.a.p.s.j.WebSocketMessageWriter] Websocket connection closed. Code: [1006], message: [WebSocket Read EOF].
2016-04-12 01:26:29,151 DEBUG [:] [] [] [] [qtp639313753-85] [i.c.p.a.p.s.j.WebSocketMessageWriter] Websocket connection closed. Code: [1006], message: [WebSocket Read EOF].
2016-04-12 01:26:29,181 DEBUG [:] [] [] [] [ecutorService-6] [c.p.e.p.i.DefaultProcessInstanceImpl] Attempting to run process [externalhandler.purge:570] on resource [20]
2016-04-12 01:26:29,184 DEBUG [54ca3d50-a78b-496b-95fd-eae516284002:570] [externalHandler:20] [externalhandler.purge] [] [ecutorService-6] [c.p.e.p.i.DefaultProcessInstanceImpl] Changing state [removed->purging] on [externalHandler:20]
2016-04-12 01:26:29,185 DEBUG [54ca3d50-a78b-496b-95fd-eae516284002:570] [externalHandler:20] [externalhandler.purge] [] [ecutorService-6] [c.p.e.p.i.DefaultProcessInstanceImpl] Exiting with code [SCHEDULED] : SCHEDULED
2016-04-12 01:26:29,187 DEBUG [:] [] [] [] [ecutorService-6] [c.p.e.p.i.DefaultProcessInstanceImpl] Exiting [SCHEDULED] process [externalhandler.purge:570] on resource [20]
2016-04-12 01:26:29,188 DEBUG [:] [] [] [] [ecutorService-6] [i.c.p.o.purge.impl.PurgeMonitorImpl ] Scheduling purge for [externalHandler] id [20]
2016-04-12 01:26:30,274 DEBUG [:] [] [] [] [ecutorService-1] [a.p.s.NonBlockingSubscriptionHandler] Got exception on write, disconnecting : class java.io.EOFException [WebSocket is closed.]
2016-04-12 01:26:30,274 DEBUG [:] [] [] [] [ecutorService-1] [i.c.p.e.memory.InMemoryEventService ] Unsubscribing from [physicalhost.create;handler=goMachineService]
2016-04-12 01:26:30,274 DEBUG [:] [] [] [] [ecutorService-1] [i.c.p.e.memory.InMemoryEventService ] Unsubscribing from [physicalhost.bootstrap;handler=goMachineService]
2016-04-12 01:26:30,274 DEBUG [:] [] [] [] [ecutorService-1] [i.c.p.e.memory.InMemoryEventService ] Unsubscribing from [physicalhost.remove;handler=goMachineService]
2016-04-12 01:26:30,274 DEBUG [:] [] [] [] [ecutorService-1] [i.c.p.e.memory.InMemoryEventService ] Unsubscribing from [ping;handler=goMachineService]
2016-04-12 01:26:31,274 DEBUG [:] [] [] [] [ecutorService-2] [a.p.s.NonBlockingSubscriptionHandler] Got exception on write, disconnecting : class java.io.EOFException [WebSocket is closed.]
2016-04-12 01:26:31,274 DEBUG [:] [] [] [] [ecutorService-2] [i.c.p.e.memory.InMemoryEventService ] Unsubscribing from [environment.finishupgrade;handler=rancher-compose-executor]
2016-04-12 01:26:31,274 DEBUG [:] [] [] [] [ecutorService-2] [i.c.p.e.memory.InMemoryEventService ] Unsubscribing from [ping;handler=rancher-compose-executor]
2016-04-12 01:26:31,274 DEBUG [:] [] [] [] [ecutorService-2] [i.c.p.e.memory.InMemoryEventService ] Unsubscribing from [environment.rollback;handler=rancher-compose-executor]
2016-04-12 01:26:31,274 DEBUG [:] [] [] [] [ecutorService-2] [i.c.p.e.memory.InMemoryEventService ] Unsubscribing from [environment.upgrade;handler=rancher-compose-executor]
2016-04-12 01:26:31,274 DEBUG [:] [] [] [] [ecutorService-2] [i.c.p.e.memory.InMemoryEventService ] Unsubscribing from [environment.create;handler=rancher-compose-executor]
```

not sure if this is related but it kills
rancher-catalog-service
rancher-compose-executor and
go-machine-service

**Comment by LLParse** [comment]
Fixed in v1.0.1-rc7
